### PR TITLE
Do not fall back to ignoring empty directories for skip when empty - Take 2

### PR DESCRIPTION
**Pull request by wolfs**
Removes the deprecation introduced in https://github.com/gradle/gradle/pull/18783 for Gradle 8.0.

Replaces #18852.

### CODE REVIEW

**Review by lptr — COMMENTED** [review]
Looks great, what should we mention in the docs?

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 1]
```diff
@@ -84,6 +84,12 @@ Warnings which became errors:
 
 - An input file collection which can't be resolved.
 
+==== Gradle does not ignore empty directories for file-trees with SkipWhenEmpty
```
Comment (lptr) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:87]:
```suggestion
==== Gradle does not ignore empty directories for file-trees with `@SkipWhenEmpty`
```

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 2]
```diff
@@ -84,6 +84,12 @@ Warnings which became errors:
...
+
+Gradle detected if an input file collection annotated with `@SkipWhenEmpty` consists only of file trees and then ignored directories automatically.
```
Comment (lptr) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:89]:
```suggestion
Previously Gradle used to detect if an input file collection annotated with `@SkipWhenEmpty` consisted only of file trees and then ignored directories automatically.
```

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 3]
```diff
@@ -84,6 +84,12 @@ Warnings which became errors:
...
+To ignore directories in Gradle 8.0 and later, the input property needs to be annotated with `@IgnoreEmptyDirectories`.
```
Comment (lptr) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:90]:
```suggestion
To ignore directories in Gradle 8.0 and later, the input property needs to be explicitly annotated with `@IgnoreEmptyDirectories`.
```

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Pull request merged**